File: assignment-2/q3/check.py
name = input("enter your: ").strip()  # Stores the user’s name
addr = input("enter email: ").strip()  # Stores the email address
matches = 0  # Whether this is a valid email address for this person
nameadd = (addr[0:addr.find("@")])
contains1 = list(name).tolower()
contains2 = list(nameadd)
freq1 = []
freq2 = []

if addr.find("@") == -1:
    print(matches)

    for i in nameadd:
        if name.find(i) == -1:
            print(matches)

    for i in range(len(contains1) - 1):
        if contains1[i] >= 'a' and contains1[i] <= 'z':
            freq1[ord(contains1[i]) - 97] = freq1[ord(contains1[i]) - 97] + 1
    for i in range(len(contains2) - 1):
        if contains2[i] >= 'a' and contains2[i] <= 'z':
            freq2[ord(contains2[i]) - 97] = freq2[ord(contains2[i]) - 97] + 1

    for i in range(26):
        if freq2[i] > freq[i]:
            print(matches)

    matches = matches + 1
    print(matches)

File: assignment-2/q3/check.c
#include <stdio.h>
#include <string.h>
#include<math.h>
int main(int x){
	char name[100];//Store the person's name
	char addr[100];//Store the email address
	int matches;//Store 0 or 1, depending on validity
	fgets(name, 100, stdin);//Find out what this does
	fgets(addr, 100, stdin);
    int atcheck = 0;
    int atpos;
    int nameArray[100];
    int addrArray[100];
	//Your code here
    for (int i = 0; addr[i] != '\0';i++){
        if (addr[i] == '@'){
            atcheck = 1;
            atpos = i;
            break;
        }else{
            matches = 0;
            atcheck = 0;
        }
    }
    if(atcheck == 1){
        for(int i = 0 ; i < atpos ; i++){
            if( (int)addr[i] >= 65 && (int)addr[i] <= 90){
                addrArray[i] = ((int)addr[i] + 32);
            }else{
                addrArray[i] = (int)addr[i];
            }
        }
        for(int i = 0 ; i < strlen(name) ; i++){
            if( (int)name[i] >= 65 && (int)name[i] <= 90){
                nameArray[i] = ((int)name[i] + 32);
            }else{
                nameArray[i] = (int)name[i];
            }
        }
        for(int i = 0 ; i < atpos ; i++){
            for(int j = 0 ; j < strlen(name);j++){
                if(addrArray[i] == nameArray[j]){
                    addrArray[i] = 0;
                    nameArray[j] = 0;
                }
            }
        }
        for(int i = 0 ; i < atpos ; i++){
            if (addrArray[i] >= 97 && addrArray[i] <= 122){
                matches = 0;
                break;
            }else{
                matches = 1;
            }
        }
    }
	printf("%d\n", matches);
	return 0;
}
